### PR TITLE
Two assert_cc cleanups

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
(Sorry for the unexpected failures, will restart the CentOS CI jobs as soon as possible)

File: src/basic/macro.h [hunk 1]
```diff
@@ -462,7 +462,8 @@ static inline int __coverity_check__(int condition) {
                  * type for the array, in the hope that checkers such as ubsan don't complain that the initializers for \
                  * the array are not representable by the base type. Ideally we'd use typeof(x) as base type, but that  \
                  * doesn't work, as we want to use this on bitfields and gcc refuses typeof() on bitfields.) */         \
-                assert_cc((sizeof((long double[]){__VA_ARGS__})/sizeof(long double)) <= 20); \
+                static const long double __assert_in_set[] __attribute__((__unused__)) = { __VA_ARGS__ }; \
```
Comment (poettering) [inline src/basic/macro.h]: you could also use our own `_unused_` macro

File: src/basic/macro.h [hunk 2]
```diff
@@ -462,7 +462,8 @@ static inline int __coverity_check__(int condition) {
...
+                assert_cc(sizeof(__assert_in_set) / sizeof(long double) <= 20); \
```
Comment (poettering) [inline src/basic/macro.h]: you could also use ELEMENTSOF

**Comment by keszybz** [comment]
Updated.

**Comment by yuwata** [comment]
@mrc0mmand https://ci.centos.org/job/systemd-pr-build-vagrant-sanitizers/1261/artifact//systemd-tmp/systemd-centos-ci/index.html is 403.

**Comment by mrc0mmand** [comment]
@yuwata yeah, that was mea culpa, pushed a fix into CentOS CI a restarted all affected jobs. Sorry for the noise.

**Comment by yuwata** [comment]
@mrc0mmand Ah, you already mentioned the issue. Sorry for the noise. Thank you.

**Pull request merged**